### DISCUSSION TITLE
Probelm in pyart.map.grid_from_radars when I created a radar object with pyart.testing.make_empty_ppi_radar

### DISCUSSION THREAD

**Discussion by gokul123kit** [ARM-DOE/pyart#1689] [Q&A] [answered]
I have a Netcdf file with radar variables with 240 grids(range dimension), 361 azimuth (dimension) and 14 elevation(Sweep) finally the reflectivity with shape (240, 361, 14)  . Then I created a radar object with using pyart.testing.make_empty_ppi_radar.  with shape 

radar = pyart.testing.make_empty_ppi_radar(240, 361, 14) 

and added the field as mentioned in "Basic ingest of a netcdf file"  https://arm-doe.github.io/pyart/notebooks/basic_ingest_using_test_radar_object.html  (But the example only contain 1 sweep( elevation))

after creating the radar object I tried to grid it with pyart.map.grid_from_radars

grid_shape  = (41, 480, 480)
grid_limits = ((0, 10000), (-120000.0, 120000.0), (-120000.0, 120000.0))
grid_roi    = 2000
```


```python
grid = pyart.map.grid_from_radars(
        [radar],
        grid_shape = grid_shape,
        grid_limits = grid_limits,
        weighting_function = 'Barnes',
        gridding_algo = 'map_gates_to_grid',
        roi_func='constant', constant_roi = grid_roi,
        fields=['reflectivity'])
```
and got the error

ValueError                                Traceback (most recent call last)

    Cell In[605], line 1
    ----> 1 grid = pyart.map.grid_from_radars(
          2         [radar],
          3         grid_shape = grid_shape,
          4         grid_limits = grid_limits,
          5         weighting_function = 'Barnes',
          6         gridding_algo = 'map_gates_to_grid',
          7         roi_func='constant', constant_roi = grid_roi,
          8         fields=['reflectivity'])
    

    File ~\AppData\Local\anaconda3\Lib\site-packages\pyart\map\grid_mapper.py:93, in grid_from_radars(radars, grid_shape, grid_limits, gridding_algo, copy_field_dtypes, **kwargs)
         91     grids = map_to_grid(radars, grid_shape, grid_limits, **kwargs)
         92 elif gridding_algo == "map_gates_to_grid":
    ---> 93     grids = map_gates_to_grid(radars, grid_shape, grid_limits, **kwargs)
         94 else:
         95     raise ValueError("invalid gridding_algo")
    

    File ~\AppData\Local\anaconda3\Lib\site-packages\pyart\map\gates_to_grid.py:169, in map_gates_to_grid(radars, grid_shape, grid_limits, grid_origin, grid_origin_alt, grid_projection, fields, gatefilters, map_roi, weighting_function, toa, roi_func, constant_roi, z_factor, xy_factor, min_radius, h_factor, nb, bsp, dist_factor, **kwargs)
        167 for i, field in enumerate(fields):
        168     fdata = radar.fields[field]["data"]
    --> 169     field_data[:, :, i] = np.ma.getdata(fdata)
        170     field_mask[:, :, i] = np.ma.getmaskarray(fdata)
        172 # find excluded gates from the gatefilter
    

    ValueError: could not broadcast input array from shape (240,361,14) into shape (5054,240)

How can I resolve this???? Gridding worked when I used one sweep but not for field with 14 sweeps

**Comment by zssherman** [comment]
@gokul123kit Hello! Question I had was, did reading in the file with read_cfradial not work? Second the reflectivity field needs to be two dimensional so in this case you would need to use the start and end sweepray index to plug in the reflectivity for each sweep. The error is most likely due to the 3 dimensions your using, were pyart expects two.

**Reply by gokul123kit** [reply]
1. my radar data is in .h5 format ('scan-ldr-120km-5_20001_20180606000836_00.h5') and could not import that in pyart, so I made .NC file with different radar data attributes and tried to make PPI from that. 

2. So should I change My reflectivity with shape (240, 361, 14) to (240, 361) ? 
   How can I give start and end sweepray index? Could you please give some information about that?

**Reply by zssherman** [reply]
@gokul123kit Did you try one of the h5 readers in Py-ART's Aux io module?

**Reply by gokul123kit** [reply]
Yes, This is the error I am getting
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
Cell In[17], line 4
      1 file_path = 'L:/Measurements/Radar/dwsr-2501c/scan-ldr-120km-5/2021/06/23/scan-ldr-120km-5_20001_20210623162850_00.h5'
      3 # Open the file with Py-ART's ODIM HDF5 reader
----> 4 radar = pyart.aux_io.read_odim_h5(file_path)

File ~\AppData\Local\anaconda3\Lib\site-packages\pyart\aux_io\odim_h5.py:164, in read_odim_h5(filename, field_names, additional_metadata, file_field_names, exclude_fields, include_fields, include_datasets, exclude_datasets, **kwargs)
    162 # metadata
    163 metadata = filemetadata("metadata")
--> 164 metadata["source"] = _to_str(hfile["what"].attrs["source"])
    165 metadata["original_container"] = "odim_h5"
    166 metadata["odim_conventions"] = _to_str(hfile.attrs["Conventions"])

File h5py\\_objects.pyx:54, in h5py._objects.with_phil.wrapper()

File h5py\\_objects.pyx:55, in h5py._objects.with_phil.wrapper()

File ~\AppData\Local\anaconda3\Lib\site-packages\h5py\_hl\attrs.py:56, in AttributeManager.__getitem__(self, name)
     52 @with_phil
     53 def __getitem__(self, name):
     54     """ Read the value of an attribute.
     55     """
---> 56     attr = h5a.open(self._id, self._e(name))
     57     shape = attr.shape
     59     # shape is None for empty dataspaces

File h5py\\_objects.pyx:54, in h5py._objects.with_phil.wrapper()

File h5py\\_objects.pyx:55, in h5py._objects.with_phil.wrapper()

File h5py\\h5a.pyx:80, in h5py.h5a.open()

KeyError: "Can't open attribute (can't locate attribute: 'source')"

**Reply by zssherman** [reply, marked as answer]
Gotcha, with your current work flow you can transpose you ref array so we can index it by sweep and plug that in by sweep like so:
```
from pyart.config import get_metadata
ref_dict = get_metadata("reflectivity")

new_ref = your_reflectivity_array.transpose(2, 1, 0)
ref = np.zeros((radar.nrays, radar.ngates))
for sweep in radar.sweep_number['data']:
    start, end = radar.get_start_end(sweep)
    ref[start:end+1, :] = new_ref[sweep]

ref_dict["data"] = ref
radar.fields = {"reflectivity": ref_dict}
```
I believe that should work...

**Reply by gokul123kit** [reply]
It worked, Thank you
